File: twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = new Map();
    let index = 0;
    for (const number of nums) {
        const diff = target - number;
        const mapped = map.get(diff);
        if (!isNaN(mapped)) {
            return mapped > index ? [index, mapped] : [mapped, index];
        } else {
            map.set(number, index);
        }
        index++;
    }
};

File: Year One/two_sum.py
from typing import List

def two_sum(nums: List[int], target: int) -> List[int]:
    differences = {}
    for index in range(len(nums)):
        if target - nums[index] not in differences:
            differences[nums[index]]=index
        else:
            return [differences[target-nums[index]],index]
